src/library/dao/directorydao.cpp

```diff
@@ -95,19 +98,23 @@ QList<RelocatedTrack> DirectoryDAO::relocateDirectory(
         return {};
     }
 
-    // on Windows the absolute path starts with the drive name
-    // we also need to check for that
-    QString startsWithOldFolder = SqlLikeWildcardEscaper::apply(
-        QDir(oldFolder).absolutePath() + "/", kSqlLikeMatchAll) + kSqlLikeMatchAll;
-
-    // Also update information in the track_locations table. This is where mixxx
-    // gets the location information for a track. Put marks around %1 so that
-    // this also works on windows
-    query.prepare(QString("SELECT library.id, track_locations.id, track_locations.location "
-                          "FROM library INNER JOIN track_locations ON "
-                          "track_locations.id = library.location WHERE "
-                          "track_locations.location LIKE '%1' ESCAPE '%2'")
-                  .arg(startsWithOldFolder, kSqlLikeMatchAll));
+    // Appending '/' is required to disambiguate files from parent
+    // directories, e.g. "a/b.mp3" and "a/b/c.mp3" where "a/b" would
+    // match both instead of only files in the parent directory "a/b/".
+    DEBUG_ASSERT(!oldFolder.endsWith('/'));
+    const QString oldFolderPrefix = oldFolder + '/';
+    const QString startsWithOldFolder = SqlLikeWildcardEscaper::apply(
```

Comment:
849af59 I wasn't aware that this does not only work for columns but also for arbitrary literals in SQL queries :eyes: 

But LIKE/GLOB wildcards still need to be escaped manually.

Reply:
The last remaining, public invocation of `SqlStringFormatter::format()` looks suspicious. I won't touch this code because I don't understand its intention.